alphadia/cli.py

```diff
@@ -135,113 +114,43 @@ def parse_config(args: argparse.Namespace) -> dict:
-        Output directory.
+def _get_raw_path_list_from_args_and_config(
+    args: argparse.Namespace, config: dict
+) -> list:
     """
+    Generate a list of raw file paths based on command-line arguments and configuration.
 
-    output_directory = None
-    if "output_directory" in config:
-        output_directory = (
-            utils.windows_to_wsl(config["output_directory"])
-            if args.wsl
-            else config["output_directory"]
-        )
-
-    if args.output is not None:
-        output_directory = (
-            utils.windows_to_wsl(args.output) if args.wsl else args.output
-        )
-
-    return output_directory
-
+    This function combines file paths specified in the configuration and command-line
+    arguments, including files from specified directories. It filters the resulting
+    list of file paths using a regular expression provided in the arguments.
 
-def parse_quant_dir(args: argparse.Namespace, config: dict) -> str:
-    """Parse custom quant path.
-    1. Use custom quant path from config file if specified.
-    2. Use custom quant path from command line if specified.
+    Args:
+        args (argparse.Namespace): Command-line arguments containing file and directory
+            paths, as well as a regex pattern for filtering.
+        config (dict): Configuration dictionary that may include a list of raw paths
+            and a directory to search for files.
 
-    Parameters
-    ----------
-
-    args : argparse.Namespace
-        Command line arguments.
-
-    config : dict
-        Config dictionary.
-
-    Returns
-    -------
-
-    quant_dir : str
-        path to quant directory.
+    Returns:
+        list: A list of file paths that match the specified regex pattern.
     """
 
-    quant_dir = None
-    if "quant_dir" in config:
-        quant_dir = (
-            utils.windows_to_wsl(config["quant_dir"])
-            if args.wsl
-            else config["quant_dir"]
-        )
-
-    if args.quant_dir is not None:
-        quant_dir = utils.windows_to_wsl(args.quant_dir) if args.wsl else args.quant_dir
-
-    return quant_dir
-
+    raw_path_list = config.get("raw_path_list", [])
+    raw_path_list += args.file
 
-def parse_raw_path_list(args: argparse.Namespace, config: dict) -> list:
-    """Parse raw file list.
-    1. Use raw file list from config file if specified.
-    2. Use raw file list from command line if specified.
-
-    Parameters
-    ----------
-
-    args : argparse.Namespace
-        Command line arguments.
-
-    config : dict
-        Config dictionary.
-
-    Returns
-    -------
-
-    raw_path_list : list
-        List of raw files.
-    """
-    config_raw_path_list = config.get("raw_path_list", [])
-    raw_path_list = (
-        utils.windows_to_wsl(config_raw_path_list) if args.wsl else config_raw_path_list
-    )
-    raw_path_list += utils.windows_to_wsl(args.file) if args.wsl else args.file
+    if (config_directory := config.get("directory")) is not None:
```

Comment:
should we add a `valid_extensions` list?

Reply:
you mean filter out all raw files that are not `.d`, `.raw`, .. ?